analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -49,27 +49,17 @@ public abstract class MethodsShouldNotHaveIdenticalImplementationsBase<TSyntaxKi
                         return;
                     }
 
-                    var methods = GetMethodDeclarations(c.Node).ToList();
+                    var methodsToHandle = new LinkedList<TMethodDeclarationSyntax>(GetMethodDeclarations(c.Node));
 
-                    var alreadyHandledMethods = new HashSet<TMethodDeclarationSyntax>();
-
-                    foreach (var method in methods)
+                    while (methodsToHandle.Count > 0)
                     {
-                        if (alreadyHandledMethods.Contains(method))
-                        {
-                            continue;
-                        }
-
-                        alreadyHandledMethods.Add(method);
-
-                        var duplicates = methods.Except(alreadyHandledMethods)
-                                                .Where(m => AreDuplicates(method, m))
-                                                .ToList();
-
-                        alreadyHandledMethods.UnionWith(duplicates);
+                        var method = methodsToHandle.First.Value;
+                        methodsToHandle.RemoveFirst();
```

Comment:
What about queue instead. 
We can't remove from queue, but we can make `duplicates` a hashset and re-create the queue from the old one `.Where(x => !duplicates.Contains(x))`

Reply:
All good points and I tried using a stack/queue also. I ended up with LinkedList because removal from the start is O(1) and that is by far the most common operation. The `remove` in the duplicates loop is rare (in orders of magnitude, because we are talking about equivalent methods) in comparison and still O(n). I think the linked list is the sweet spot here because removal from the start is the most important thing.